lib/resque/plugins/round_robin/round_robin.rb

```diff
@@ -17,12 +17,15 @@ def self.configure
     end
 
     def rotated_queues
+      @priority_order ||= @queues.map.with_index {|q, i| [q.chomp("*"), i] }.to_h
       @n ||= 0
       @n += 1
       rot_queues = queues # since we rely on the resque-dynamic-queues plugin, this is all the queues, expanded out
       if rot_queues.size > 0
         @n = @n % rot_queues.size
-        rot_queues.rotate(@n)
+        rot_queues.rotate(@n).sort_by do |queue|
+          @priority_order[queue.sub(/\d+$/, '')] || @priority_order.size
```

Comment:
This logic works for our specific needs, but not in the general case if we want to contribute this change back to the community. Not everyone will have queue names ending with digits.

What do you think about doing something like this instead?

```suggestion
          @priority_order.find { |k, v| queue.start_with?(k) }&.last || @priority_order.size
```

The `@priority_order` hash should be small enough that it won't be much of a performance hit.

Is there a reason why `@priority_order.size` is the fallback value? Just curious.

Reply:
I just thought about it some more and realized it's the fallback so that in the unlikely event that resque-dynamic-queues returns a queue name that doesn't match a `@priority_order` hash key, it will be prioritized last.

Reply:
Yeah that makes sense, I'll add that change in. 

The default of `@priority_order.size` is not intended to really be hit, but if, for whatever reason, there are queues in `rot_queues` that don't match the priority queues, the fallback allows them to be slotted into the end. Without the fallback, it will be comparing the indexes to `nil` and it would throw an exception. So it is more of a safety guard.